docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,277 @@
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Resumed"
+    }"
+```
+
+**Prerequisites**: You can resume the microVM any time after the microVM is booted.
+                   Successive calls of this request keep the microVM in the `Resumed`
+                   state.
+**Effects**:
+- _on success_: microVM is guaranteed to be `Resumed`.
+- _on failure_: no side-effects.
+
+## Loading snapshots
+
+If you want to load a snapshot, you can do that only **before** the microVM is configured
+(the only resource that can be configured prior is the Logger) by sending the following
+API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/load" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"enable_diff_snapshots\": true
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: Does **not** work with diff snapshots.
+**Effects:**
+- _on success_:
+  - The complete microVM state will be loaded from snapshot into the current Firecracker
+    process.
+  - It resets then the dirtied page bitmap and marks all pages clean (from an incremental
+    snapshot point of view).
+  - The loaded microVM is now in the `Paused` state and needs to be resumed for it to run.
+  - The file indicated by `mem_file_path` from which this microVM was resumed
+    **must** not change throughout the lifetime of the microVM. Since the guest in the
+    microVM will load pages from the file on demand, the Firecracker process should be
+    considered to have a shared-read-lock on the file. Any external writes to the file can
+    lead to microVM guest *Undefined Behavior*.
+  - The file indicated by `snapshot_path`, that is used to load from, is released and no
+    longer used by this process.
+  - If `enable_diff_snapshots` is set, then incremental snapshots can be taken afterwards.
+- _on failure_: The encountered error is reported, then the current Firecracker process
+                is ended (as it might be in an invalid state).
+
+Please, keep in mind that only if you set to true `enable_diff_snapshots`, when loading a
+snapshot, or `track_dirty_pages`, when configuring the machine on a fresh microVM, you can
+then create a `diff` snapshot. Also, `track_dirty_pages` is not saved when creating a
+snapshot, so you need to explicitly set `enable_diff_snapshots` when sending `LoadSnapshot`
+command if you want to be able to do diff snapshots from a loaded microVM.
+Another thing that you should be aware of is the following: if a fresh microVM can create
+incremental snapshots, then if you create a **full** snapshot, the memory file contains
```

Comment:
Please use `diff` snapshots instead of `incremental`

Reply:
I removed all `incremental` occurrences.